Jar indexing fails due to unreachable instructions for xmlpull

Affected jar (most likely more): <https://repo1.maven.org/maven2/xmlpull/xmlpull/1.1.3.4d_b4_min/xmlpull-1.1.3.4d_b4_min.jar>
Affected version(s): 3.0.1 (possibly earlier), current master

```
java.lang.RuntimeException: java.lang.RuntimeException: Exception while indexing class: org/xmlpull/mxp1/MXParser
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.base/java.lang.reflect.Constructor.newInstanceWithCaller(Constructor.java:500)
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:481)
	at java.base/java.util.concurrent.ForkJoinTask.getThrowableException(ForkJoinTask.java:562)
	at java.base/java.util.concurrent.ForkJoinTask.reportException(ForkJoinTask.java:591)
	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:689)
	at java.base/java.util.stream.ForEachOps$ForEachOp.evaluateParallel(ForEachOps.java:159)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateParallel(ForEachOps.java:173)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.base/java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:596)
	at java.base/java.util.stream.ReferencePipeline$Head.forEach(ReferencePipeline.java:765)
	at cuchaz.enigma.analysis.index.JarIndex.indexJar(JarIndex.java:89)
	at cuchaz.enigma.Enigma.openJars(Enigma.java:85)
	at cuchaz.enigma.gui.GuiController.lambda$openJar$1(GuiController.java:141)
	at cuchaz.enigma.gui.dialog.ProgressDialog.lambda$runOffThread$1(ProgressDialog.java:97)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:718)
	at java.base/java.util.concurrent.CompletableFuture$Completion.exec(CompletableFuture.java:483)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:373)
	at java.base/java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec(ForkJoinPool.java:1182)
	at java.base/java.util.concurrent.ForkJoinPool.scan(ForkJoinPool.java:1655)
	at java.base/java.util.concurrent.ForkJoinPool.runWorker(ForkJoinPool.java:1622)
	at java.base/java.util.concurrent.ForkJoinWorkerThread.run(ForkJoinWorkerThread.java:165)
Caused by: java.lang.RuntimeException: Exception while indexing class: org/xmlpull/mxp1/MXParser
	at cuchaz.enigma.analysis.index.JarIndex.lambda$indexJar$1(JarIndex.java:93)
	at java.base/java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:183)
	at java.base/java.util.Spliterators$ArraySpliterator.forEachRemaining(Spliterators.java:992)
	at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509)
	at java.base/java.util.stream.ForEachOps$ForEachTask.compute(ForEachOps.java:290)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:754)
	... 5 more
Caused by: java.lang.NullPointerException: Cannot invoke "java.util.List.size()" because "this.stack" is null
	at cuchaz.enigma.analysis.index.IndexReferenceVisitor$IndexReferenceMethodVisitor.getReferenceTargetType(IndexReferenceVisitor.java:142)
	at cuchaz.enigma.analysis.index.IndexReferenceVisitor$IndexReferenceMethodVisitor.visitFieldInsn(IndexReferenceVisitor.java:60)
	at org.objectweb.asm.tree.FieldInsnNode.accept(FieldInsnNode.java:88)
	at org.objectweb.asm.tree.InsnList.accept(InsnList.java:144)
	at org.objectweb.asm.tree.MethodNode.accept(MethodNode.java:749)
	at org.objectweb.asm.tree.MethodNode.accept(MethodNode.java:647)
	at org.objectweb.asm.tree.ClassNode.accept(ClassNode.java:468)
	at cuchaz.enigma.analysis.index.JarIndex.lambda$indexJar$1(JarIndex.java:91)
	... 10 more
```

--- a/enigma/src/main/java/cuchaz/enigma/analysis/index/IndexReferenceVisitor.java
+++ b/enigma/src/main/java/cuchaz/enigma/analysis/index/IndexReferenceVisitor.java
@@ -139,7 +139,7 @@ public class IndexReferenceVisitor extends ClassVisitor {
 		}
 
 		private ReferenceTargetType getReferenceTargetType(int stackDepth) {
-			if (stack == null) { // Unreachable instruction if ASM is to be trusted
+			if (stack == null) { // Unreachable instruction
 				return ReferenceTargetType.uninitialized();
 			}
 

